File: packages/python/examples/usage.py
from llm_local_client import LLMClient
import json

def run():
    client = LLMClient(model='llama3')
    
    try:
        print('Initializing client...')
        client.initialize()
        print('Client initialized.')

        print('Sending prompt 1...')
        response1 = client.prompt(message='Explain what a DAG is in 1 sentence.')
        print(f'Assistant: {response1}\n')

        # Get the context of the last interaction to branch from it
        branches = client.get_context().get_conversation_branches()
        last_node_id = branches[0][-1] # The assistant's response node

        print('Sending prompt 2 (follow up)...')
        response2 = client.prompt(
            message='Give me a real world example of it.',
            context_id=last_node_id
        )
        print(f'Assistant: {response2}\n')

        print('Full conversation tree:')
        nodes = [{"id": n.id, "role": n.role, "content": n.content} for n in client.get_context().node_list()]
        print(json.dumps(nodes, indent=2))

    except Exception as e:
        print(f'Error during execution: {e}')

if __name__ == '__main__':
    run()

File: packages/typescript/examples/usage.ts
import { LLMClient } from '../src';

async function run() {
  const client = new LLMClient({ model: 'llama3' });
  
  try {
    console.log('Initializing client...');
    await client.initialize();
    console.log('Client initialized.');

    console.log('Sending prompt 1...');
    const response1 = await client.prompt({ message: 'Explain what a DAG is in 1 sentence.' });
    console.log(`Assistant: ${response1}\n`);

    // Get the context of the last interaction to branch from it
    const branches = client.getContext().getConversationBranches();
    const lastNodeId = branches[0][branches[0].length - 1]; // The assistant's response node

    console.log('Sending prompt 2 (follow up)...');
    const response2 = await client.prompt({ 
      contextId: lastNodeId, 
      message: 'Give me a real world example of it.' 
    });
    console.log(`Assistant: ${response2}\n`);

    console.log('Full conversation tree:');
    console.log(JSON.stringify(client.getContext().nodeList(), null, 2));

  } catch (error) {
    console.error('Error during execution:', error);
  }
}

run();
